lua/bufferline/layout.lua

```diff
@@ -70,10 +71,19 @@ function Layout.calculate_buffer_width(bufnr, index, use_buffer_index, use_file_
     end
 
     if use_file_icon then
+      --- @diagnostic disable-next-line:param-type-mismatch
       local file_icon = icons.get_icon(bufnr, '')
       width = width + strwidth(file_icon) + 1 -- icon + space after icon
     end
 
+    if diagnostics.enable then
```

Comment:
This is where the diagnostic symbols are accounted for when calculating how much space each buffer takes in the bufferline.